### DISCUSSION TITLE
Question about missing GUID/CLSID's in EarTrumpet Code

### DISCUSSION THREAD

**Discussion by ZKAT8IT** [File-New-Project/EarTrumpet#1342] [Q&A] [answered]
Hello, I'm currently developing a plugin for another application and was looking through your code and noticed you had a method for changing the output audio device of a specific application, located in your IAudioPolicyConfigFactoryVariantFor21H2 class/interface. However, I do not seem to see the GUID/CLSID of "ab3d4648-e242-459f-b02f-541c70306324" or "2a59116d-6c4f-45e0-a74f-707e3fef9258" for the downlevel variant on my system even after installing the Windows SDK and .Net SDK. What DLL's do these reference to and where can I get them?

**Comment by riverar** [comment, marked as answer]
Hi @ZKAT8IT, these are undocumented WinRT interfaces/classes, reverse engineered from the OS provided audioses.dll.
